platform/mds/mds/src/main/java/org/motechproject/mds/domain/Entity.java

```diff
@@ -641,6 +641,9 @@ public void updateIndexes(List<LookupDto> indexes) {
                 }
                 String lookupFieldName = LookupName.buildLookupFieldName(field.getName(), lookupField.getRelatedName());
                 lookupFieldsOrder.add(lookupFieldName);
+                if(lookupField.getName() != field.getName()) {
```

Comment:
Moreover, you should not use != and == for comparing strings, see http://stackoverflow.com/questions/767372/java-string-equals-versus